posts/2024-08-20-liberty-olo-rh-sso.adoc

```diff
@@ -0,0 +1,338 @@
+
+. Test the user you created.
++
+Go to the `\https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/Sso-realm/console/` URL and log in by using the `testuser1`/`testpasswd1` username you created in the previous steps. After you log in, in the **General** section, you can see the endpoints.
+. Click the link for the **OpenID Endpoint Configuration**, which points you to the following URL:
++
+`\https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration`.
++
+You need this URL as the `discoveryEndpoint` for the client registration later on.
+
+[#createSecret]
+== Create the olapp-sso secret
+
+Next, we create a secret for the Open Liberty application. The key name for this secret is in the `<app-name>-olapp-sso` format. You must use the same `<app-name>` as the one you use to deploy the application with OLO. For example, in the application that is used in this post, the `<app-name>` from the YAML file is `guide-social-media-login`, so the secret name is `guide-social-media-login-olapp-sso`.
+
+. Create a project in your cluster called `gsm-test`.
+. Click **Workloads** > **Secrets** >** Create Secret**, a create a secret that is called `guide-social-media-login-olapp-sso`. Specify `oidc-clientId` as the **Key** and `gsmapp` as the **Value**.
+
+[.img_border_light]
+image::img/blog/rh_create_secret.png[Create olapp-sso secret,width=50%,align="center"]
+
+
+[#deployApp]
+== Install the Open Liberty Operator and deploy the application to Red Hat OpenShift
+
+If the Open Liberty Operator (OLO) isn't already installed in your OCP cluster, install it by following the link:https://github.com/OpenLiberty/open-liberty-operator/blob/main/doc/user-guide-v1.adoc[user guide].
+
+After installing the Open Liberty Operator, use the following YAML file to deploy the `guide-social-media-login` application that you previously created.
+
+[source]
+----
+apiVersion: apps.openliberty.io/v1
+kind: OpenLibertyApplication
+metadata:
+  name: guide-social-media-login
+  namespace: gsm-test
+spec:
+  sso:
+    oidc:
+      - discoveryEndpoint:
+          <https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration>
+  service:
+    port: 9443
+  applicationImage:
+    <image location of the app>
+  expose: true
+  manageTLS: true
+  replicas: 1
+  applicationName: guide-sm-login
+  pullPolicy: Always
+  pullSecret: <secret_to_pull_image>
+
+
+----
+
+* The name of the application is `guide-social-media-login`, the same name that was used when you created the previous secret.
+* The `oidc: discoveryEndpoint` must point to the OpenID endpoint configuration that you set when you configured the RH-SSO Operator.
+* The `applicationImage` value must point to your image location where you placed your application image, in a container registry like DockerHub or Artifactory.
+* The `pullSecret` value must be set to be able to access the container registry.
```

Comment:
> The `pullSecret` value must be set to be able to access the container registry.

what should it be set to- is it the `guide-social-media-login-olapp-sso` secret they created in the previous step?

Reply:
No, this secret is different.
After you upload your container image (containing your application) to a container registry, besides pointing to the url that has the image, you also need to provide it the credentials to be able to access that registry.

In my case, I was using Artifactory to push the image to, and I have my pull-secret pointing to the credentials that I use to log into Artifactory.

This part is more generic, because we can use any type of container registry (such as dockerhub etc). But basically after you point to the image, you also have to provide the credentials that OCP will need to pull down that image.